src/app/shared/utils/data.utils.ts

```diff
@@ -86,7 +87,14 @@ export function formatDateTimeDataValue(value: any, config: DateTimeConstraintCo
 
 export function formatNumberDataValue(value: any, config: NumberConstraintConfig): string {
   // TODO format based on config
-  return formatUnknownDataValue(value);
+  if ([undefined, null, ''].includes(value)) {
+    return '';
+  }
+  const valueBig = convertToBig(value);
+  if (valueBig) {
+    return decimalStoreToUser(valueBig.toFixed());
+  }
+  return String(value);
```

Comment:
All formatting functions should by default use `formatUnknownDataValue` for values they cannot process.